### ISSUE TITLE
A possible error

### ISSUE THREAD

**Issue by pilgrimygy** [JuliaReinforcementLearning/ReinforcementLearningZoo.jl#154]
I found a possible error:
In rainbow.jl, calculating the error of lines 177 to 179, the logits from the online network don't use the "softmax" operation.

**Comment by findmyway** [comment]
> In rainbow.jl, calculating the error of lines 177 to 179, the logits from the online network don't use the "softmax" operation.

You are correct here. But I think the `target_distribution` in line 179 is also not normalized to sum to 1.

https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/c5fa62828dc39ba3824a55cf259ffdf187289251/src/algorithms/dqns/rainbow.jl#L179

So now whether it is a bug or not mainly depends on how you define the `loss_func` here.
A typical one is the `logitcrossentropy`:

https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/c5fa62828dc39ba3824a55cf259ffdf187289251/src/experiments/atari/Dopamine_Rainbow_Atari.jl#L54

But I think the doc string doesn't make it clear here. Wanna make a PR to update it here? 😀 

https://github.com/JuliaReinforcementLearning/ReinforcementLearningZoo.jl/blob/c5fa62828dc39ba3824a55cf259ffdf187289251/src/algorithms/dqns/rainbow.jl#L12

**Comment by pilgrimygy** [comment]
OK. I will try whether the softmax operation has an impact on performance. 
I am glad to add some explanation in the doc.

**Comment by pilgrimygy** [comment]
12 lines of rainbow.jl:
```
- `loss_func`: the loss function. It is recommended to use Flux.Losses.logitcrossentropy. Flux.Losses.crossentropy will encounter the problem of negative numbers.
```
177-179 lines of rainbow.jl
```
logits = reshape(Q(states), n_atoms, n_actions, :)
select_logits = logits[:, actions]
# The original paper normalized logits, but using normalization and Flux.Losses.crossentropy is not as stable as using Flux.Losses.logitcrossentropy.
batch_losses = loss_func(select_logits, target_distribution)
```

**Comment by findmyway** [comment]
How about making a PR to help others aware of it? 😉
